src/test/java/com/googleinterns/zoomtube/utils/CommentUtilTest.java

```diff
@@ -46,67 +49,96 @@ public void tearDown() {
+    assertThat(comment.type()).isEqualTo(type);
   }
 
   @Test
-  public void createEntity_noParent_shouldReturnEntityWithProperties_noParent() throws IOException {
-    Key lectureKey = KeyFactory.createKey(LectureUtil.KIND, 12345);
+  public void createComment_root_shouldReturnCommentFromEntity() throws IOException {
+    long timestamp = 123;
+    User author = new User("test@example.com", "example.com");
+    String content = "Test content";
+    Date dateNow = new Date();
+    Entity entity = new Entity(CommentUtil.KIND);
+    Comment.Type type = Comment.Type.QUESTION;
+    entity.setProperty(CommentUtil.LECTURE, lectureKey);
+    entity.setProperty(CommentUtil.TIMESTAMP_MS, timestamp);
+    entity.setProperty(CommentUtil.AUTHOR, author);
+    entity.setProperty(CommentUtil.CONTENT, content);
+    entity.setProperty(CommentUtil.CREATED, dateNow);
+    entity.setProperty(CommentUtil.TYPE, type.toString());
+
+    Comment comment = CommentUtil.createComment(entity);
+
+    assertThat(comment.commentKey()).isEqualTo(entity.getKey());
+    assertThat(comment.lectureKey()).isEqualTo(lectureKey);
+    assertThat(comment.parentKey().isPresent()).isFalse();
+    assertThat(comment.timestampMs().isPresent()).isTrue();
+    assertThat(comment.timestampMs().get()).isEqualTo(timestamp);
+    assertThat(comment.author()).isEqualTo(author);
+    assertThat(comment.content()).isEqualTo(content);
+    assertThat(comment.created()).isEqualTo(dateNow);
+    assertThat(comment.type()).isEqualTo(type);
+  }
+
+  @Test
+  public void createRootEntity_shouldReturnEntityWithProperties() throws IOException {
     long timestampMs = 123;
     User author = new User("test@example.com", "example.com");
     String content = "Test content";
     Date dateNow = new Date();
+    Comment.Type type = Comment.Type.QUESTION;
 
-    Entity entity = CommentUtil.createEntity(lectureKey, timestampMs, author, content, dateNow);
+    Entity entity =
+        CommentUtil.createRootEntity(lectureKey, timestampMs, author, content, dateNow, type);
 
     assertThat(entity.getProperty(CommentUtil.LECTURE)).isEqualTo(lectureKey);
     assertThat(entity.getProperty(CommentUtil.TIMESTAMP_MS)).isEqualTo(timestampMs);
+    assertThat(entity.getProperty(CommentUtil.PARENT)).isNull();
     assertThat(entity.getProperty(CommentUtil.AUTHOR)).isEqualTo(author);
     assertThat(entity.getProperty(CommentUtil.CONTENT)).isEqualTo(content);
     assertThat(entity.getProperty(CommentUtil.CREATED)).isEqualTo(dateNow);
+    assertThat(Comment.Type.valueOf((String) entity.getProperty(CommentUtil.TYPE))).isEqualTo(type);
   }
 
   @Test
-  public void createEntity_withParent_shouldReturnEntityWithProperties() throws IOException {
-    Key lectureKey = KeyFactory.createKey(LectureUtil.KIND, 12345);
+  public void createReplyEntity_shouldReturnEntityWithProperties() throws IOException {
     Key parentKey = KeyFactory.createKey(CommentUtil.KIND, 67890);
```

Comment:
For all raw arguments you need to add comments because idk what 67890 means. Please look at: (Document ambiguous literal arguments at the call site) https://docs.google.com/document/d/1ZQz9VulpweKEbg28epLO_tZhCboRF1DtWLrVZZlUYBE/edit#heading=h.48miiab4h6ve

Reply:
Added